imports/ui/components/modals/editProfile.js

```diff
@@ -1,55 +1,37 @@
-/* eslint-env browser */
-
 import { Template } from 'meteor/templating'
-import '/imports/api/users.js'
+import { showModal, hideModal } from '/imports/lib/utils.js'
+import '/imports/ui/components/modals/editProfileInfo.js'
+import '/imports/ui/components/modals/editPassword.js'
 import './editProfile.html'
 
+let modalToShow = null
+
+Template.editProfile.onRendered(function () {
+  this.$('.main-modal').on('hidden.bs.modal', function (e) {
```

Comment:
This is necessary so that we can trigger a modal from another modal. It may be good to, in the future, move this kind of logic to a shared place so that it will not have to be re-written for multiple templates.

Essentially, we have to wait until the previous modal has been completely hidden before attempting to show the next modal, because you cannot render two at once. There is a a `Modal.allowMultiple` attribute that does, indeed, allow us to get around this particular issue but it causes a heap of other issues that I would like to avoid :)

Reply:
The `showModal` function should already handle this: it opens a new modal if now modal is open, otherwhise it will replace the modal contents inline.

Reply:
I can take a look at this, but it did not _seem_ to work out of the box. Could have been my error!

Reply:
Actually, I think I know why it did not work. The first modal is not using the `showModal` util because it uses a fairly different design (no title, no X button, positioning):

![image](https://user-images.githubusercontent.com/7697924/32661163-d17b7b62-c5f3-11e7-9479-96c4cf9e27bf.png)

So if we are triggering a `mainModal` from another `mainModal` it should work, but not in this case. Best solution may be to switch the modal shown above to use `mainModal` and allow it to accommodate the differences in this particular modal.